ui/src/app/utils/machines.ts

```diff
@@ -45,5 +51,92 @@ export const allMachineTypes = fp.map(({ price, preemptiblePrice, ...details })
 // See https://broadworkbench.atlassian.net/browse/SATURN-1337
 export const validLeonardoMachineTypes = allMachineTypes.filter(({memory}) => memory >= 4);
 
+export const findMachineByName = (machineToFind: string) => fp.find(({name}) => name === machineToFind, allMachineTypes);
+
 export const diskPrice = 0.04 / 730; // per GB hour, from https://cloud.google.com/compute/pricing
 export const dataprocCpuPrice = 0.01; // dataproc costs $0.01 per cpu per hour
+
+export const machineStoragePrice = ({masterDiskSize, numberOfWorkers, workerDiskSize }) => {
+  if (numberOfWorkers && workerDiskSize) {
+    return (masterDiskSize + numberOfWorkers * workerDiskSize) * diskPrice;
+  } else {
+    return masterDiskSize * diskPrice;
+  }
+};
+
+export const machineStorageCostBreakdown = ({masterDiskSize, numberOfWorkers, workerDiskSize}) => {
+  const costs = [];
+  if (numberOfWorkers && workerDiskSize) {
+    costs.push(`${formatUsd(masterDiskSize * diskPrice)}/hr Master Disk`);
+    costs.push(`${formatUsd((numberOfWorkers * workerDiskSize) * diskPrice)}/hr Worker Disks`);
+  } else {
+    costs.push(`${formatUsd(masterDiskSize * diskPrice)}/hr Disk`);
+  }
+  return costs;
+};
+
+export const machineRunningPrice = ({
+  computeType,
+  masterDiskSize,
+  masterMachineName,
+  numberOfWorkers = 0,
+  numberOfPreemptibleWorkers = 0,
+  workerDiskSize,
+  workerMachineName
+}) => {
+  const masterMachine = findMachineByName(masterMachineName);
+  const workerMachine = workerMachineName && findMachineByName(workerMachineName);
+
+  if (computeType === ComputeType.Dataproc && !workerMachine) {
```

Comment:
While our UI doesn't currently support 0 node Dataproc clusters, they are legal, so I would probably handle this case